src/theory/arith/nl/pow2_solver.cpp

```diff
@@ -148,6 +152,17 @@ void Pow2Solver::checkFullRefine()
         }
     }
 
+    // triviality lemmas: pow2(x) = 0 whenever
+    // x < 0
+    if (x < 0)
+    {
+      Node assumption = nm->mkNode(LT, n[0], d_zero);
```

Comment:
Can you guard this so that it only applies when the premise is true in the model?

Reply:
done